### DISCUSSION TITLE
Custom function, how to merge duplicate records ?

### DISCUSSION THREAD

**Discussion by apcloic** [YetiForceCompany/YetiForceCRM#16178] [How to] [answered]
Hi,

I do regular import of contacts from a google spreadsheet using a custom script.
When importing, I check for existing records based on some fields using a custom function.
If a record is found, I update the record, else I create a new one.
Sometimes I find more than one existing record and so I would like to merge those records before updating it.
Does anybody knows what is the proper way to call the merge record function ?
Regards,

**Comment by vovpff** [comment]
@apcloic I think at first you need to parse your source spreadsheet to check for duplicates. In case there is duplicate you need to merge data by php concatenation. When file done to import you need to use your custom function to import prepaired data into CRM

**Reply by apcloic** [reply]
@vovpff Thanks Vladimir, I think I was not clear enough ;) 
Using my script, I already know how to parse my source spreadsheet, check and merge if I find duplicates entries.
However I would like to know if there is a way and how to call the Merging Records function, the one that you can find in contact ListView / Action menu ?
![image](https://user-images.githubusercontent.com/2930560/174609302-56dca8b0-936d-452b-8fbd-50e27b08a2ac.png)

Regards,

**Comment by rskrzypczak** [comment, marked as answer]
Here: https://github.com/YetiForceCompany/YetiForceCRM/blob/41c0b90e17c859bb0e23c921b4d03b0a4cb1ed45/modules/Vtiger/actions/MergeRecords.php#L43-L48
